staging/src/k8s.io/code-generator/generate-groups.sh

```diff
@@ -25,7 +25,7 @@ if [ "$#" -lt 4 ] || [ "${1}" == "--help" ]; then
   cat <<EOF
 Usage: $(basename "$0") <generators> <output-package> <apis-package> <groups-versions> ...
 
-  <generators>        the generators comma separated to run (deepcopy,defaulter,applyconfiguration,client,lister,informer) or "all".
```

Comment:
it's so squirrelly defaulter-gen is in generate-internal-groups... logically, defaulting belongs in the external group generator, and doesn't need the internal types *at all* (defaulting only looks at the external type, and is per-version)

Reply:
I could flip it around so internal call non-internal.  The point is that there's no real reason to have both and these scripts should EOL:

I have more commits queued up, some choice ones:
https://github.com/kubernetes/kubernetes/commit/692adf02d8c35554e0a8838ea32c93837f50796e
https://github.com/kubernetes/kubernetes/commit/3a013a771fd0a5dc2d37a9495acf4cbb3e8de5ed

See the whole branch if you want, I was just rate-limiting how many PRs I send you and end up needing to rebase :)

https://github.com/kubernetes/kubernetes/compare/master...thockin:kubernetes:codegen_new_generate_groups_approach

Reply:
Ah, just got to the end of the stack, if this is a somewhat intermediate state and it makes more sense in the end, I don't especially care, was just noting things that looked weird.

it's not like this is *removing* defaulter-gen from the external script, so I don't object to making what this *does* do explicit